Sources/News.php

```diff
@@ -1849,4 +2129,758 @@ function getXmlProfile($xml_format)
 	return $data;
 }
 
+/**
+ * Get a user's posts.
+ * The returned array will be generated to match the xml_format.
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPosts($xml_format)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	$uid = isset($_GET['u']) ? (int) $_GET['u'] : $user_info['id'];
+
+	if (empty($uid) || (!allowedTo('profile_view') && $uid != $user_info['id']))
+		return array();
+
+	$show_all = $user_info['is_admin'] || $uid == $user_info['id'];
+
+	// You are allowed in this special case to see your own posts from anywhere
+	if ($show_all)
+		$query_this_board = preg_replace('/\{query_see_board\}\s*(AND )?/', '', $query_this_board);
+
+	require_once($sourcedir . '/Subs-Attachments.php');
+
+	// Need to know the total so we can track our progress
+	if (!empty($context['batch_mode']) && empty($context['batch_total']))
+	{
+		$request = $smcFunc['db_query']('', '
+			SELECT COUNT(m.id_msg)
+			FROM {db_prefix}messages as m
+				INNER JOIN {db_prefix}boards AS b ON (b.id_board = m.id_board)
+			WHERE id_member = {int:uid}
+				AND ' . $query_this_board . ($modSettings['postmod_active'] && !$show_all ? '
+				AND approved = {int:is_approved}' : ''),
+			array(
+				'uid' => $uid,
+				'is_approved' => 1,
+			)
+		);
+		list($context['batch_total']) = $smcFunc['db_fetch_row']($request);
+		$smcFunc['db_free_result']($request);
+	}
+
+	$request = $smcFunc['db_query']('', '
+		SELECT m.id_msg, m.id_topic, m.id_board, m.poster_name, m.poster_email, m.poster_ip, m.poster_time, m.subject,
+			modified_time, m.modified_name, m.modified_reason, m.body, m.likes, m.approved, m.smileys_enabled
+		FROM {db_prefix}messages as m
+			INNER JOIN {db_prefix}boards AS b ON (b.id_board = m.id_board)
+		WHERE id_member = {int:uid}
+			AND id_msg > {int:start_after}
+			AND ' . $query_this_board . ($modSettings['postmod_active'] && !$show_all ? '
```

Comment:
While testing i got an error because $query_this_board was empty which resulted in a invalid sql query.